crypto/bn/bn_dh.c

```diff
@@ -10,7 +10,6 @@
 #include "bn_local.h"
 #include "internal/nelem.h"
 
-#ifndef OPENSSL_NO_DH
 # include <openssl/dh.h>
```

Comment:
Why are these constants not DH specific??

Reply:
(me, I've always wondered why they are defined in the BN sub-library...)

Reply:
What I've always been thinking is that if BN was a separate, general purpose bignum project, we would *never* ask such a project to please insert *our* numbers into their code.  Now, that's not quite our reality, but if we're thinking in terms of layering and generality, have DH numbers in the BN sub-library is simply strange.

Reply:
Well perhaps. But the fact is that at least some of these are exposed via BN_* calls. You can argue the rights and wrongs of this (and I think there are arguments to be made either way). But, no matter, I don't really view this as in scope of this PR (which is about removing low-level DH use in libssl) to refactor the BN<->DH layering (nor do I think that needs to be done for 3.0)

Reply:
Yes, sorry, agreed, it's off-topic for this PR